packages/telemetry/src/slog-to-otel.js

```diff
@@ -347,110 +407,167 @@ export const makeSlogToOtelKit = (tracer, overrideAttrs = {}) => {
+          const rejects = [];
+          for (const n of notifications) {
+            const [kpid, { state }] = n;
+            if (state === 'rejected') {
+              rejects.push(kpid);
+            } else {
+              resolves.push(kpid);
+            }
+          }
+          const detailses = [];
+          if (resolves.length) {
+            detailses.push(`resolve(${resolves.join(',')})`);
+          }
+          if (rejects.length) {
+            detailses.push(`reject(${rejects.join(',')})`);
+          }
+          const details = `d-${attrs.vatID}: ${detailses.join(', ')}`;
+          spans.get(getCrankKey()).setAttributes(cleanAttrs({ details }));
+
           // Track call graph.
-          const [_type, [notification]] = kd;
           // TODO: We only track the first notified kernel promise. Perhaps we should do more?
-          const [kpId, { state }] = notification;
-          const error = state === 'rejected' ? 'rejected' : undefined;
-          if (kernelPromiseToSendingCause.has(kpId)) {
+          const [tKpid, { tState }] = notifications[0];
+          const error = tState === 'rejected' ? 'rejected' : undefined;
+
+          // TODO: figure out if this is correct?
+          if (kernelPromiseToSendingCause.has(tKpid)) {
             // Track this notification as the cause for the current crank.
-            const cause = JSON.parse(kernelPromiseToSendingCause.get(kpId));
+            const cause = JSON.parse(kernelPromiseToSendingCause.get(tKpid));
+            // crankNum doesn't exist on replay
             crankNumToCause.init(slogAttrs.crankNum, {
               ...cause,
-              attrs: { ...cause.attrs, state },
+              attrs: { ...cause.attrs, state: tState },
               error,
             });
           }
         } else if (kd[0] === 'message') {
-          // This is where the message is delivered.
-          // Track call graph.
           const [
             _type,
             target,
             { methargs, method = extractMethod(methargs), result },
           ] = kd;
+          const details = `d-${attrs.vatID}: E(${target}).${method}`;
+          const msgAttrs = {
+            'msg.target': target,
+            'msg.method': extractMethod(methargs),
+            'msg.slots': methargs.slots.join(','),
+            'msg.result': result,
+            details,
+          };
+          spans
+            .get(getCrankKey())
+            .setAttributes(cleanAttrs({ ...attrs, ...msgAttrs }));
+          // This is where the message is delivered.
+          // Track call graph.
           let cause;
           if (kernelPromiseToSendingCause.has(result)) {
             cause = JSON.parse(kernelPromiseToSendingCause.get(result));
           } else {
             // Create a new root span.
-            const name = `E(${target}).${method}`;
+            const crankSpan = spans.get(getCrankKey());
             cause = {
-              name,
+              name: `deliver-${kd[0]}`,
```

Comment:
Actually, if we were to rename the `message` delivery to something else (e.g., `msg` or `communique`, say), this would _prevent_ that renaming from breaking the code.

Reply:
Right from that angle. I was more concerned about a refactor in this file. It's already pretty obscure as-is ;)